cart.mjs

```diff
@@ -1,14 +1,117 @@
 import {carts, products, promoCode, users} from "./dataset.mjs";
 import * as core from "./core-cart.mjs";
 
-console.log('Promo', promoCode);
-console.log('Products', products);
-console.log('Users', users);
-console.log('Cart', carts);
+//console.log('Promo', promoCode);
+//console.log('Products', products);
+//console.log('Users', users);
+//console.log('Cart', carts);
 
-console.log('Discount', core.discountedPrice(100, 0.2));
+//console.log('Discount', core.discountedPrice(100, 0.2));
+var dir = './receipts';
+
+if (!core.fs.existsSync(dir)){
+        core.fs.mkdirSync(dir);
+    }
 
 for (let cartRow of carts) {
+/*
 
+    let UUIDCorrente=cartRow.user;
+    let prodottiUtente=cartRow.products;
+    //console.log(UUIDCorrente);
     // genera ricevuta
-}
\ No newline at end of file
+    
+    let user = core.getUser(cartRow.user);
+
+
+
+    console.log(user,'\n');
+    console.log(prodottiUtente,'\n');*/
+
+    let user=core.getUser(cartRow.user);
+    let uuid=user.uuid;
+    let promo=user.promo;
+    let sconto=core.getPercentageFromPromoCode(promo)
+    let prodottiUtente=cartRow.products;
+    let nome=user.firstName;
+    let cognome=user.lastName;
+    let scontoEffettivo;
+    let totaleScontato;
+    let rimanenza;
+    let ricevuta='';
+
+
+
+   if (prodottiUtente.length>0){
+        let portafoglio=user.wallet;
+        let totale=0;
+        for (let oggetta of prodottiUtente){
+            let oggetto=core.getProduct(oggetta);
+            let prezzo=oggetto.price;
+            totale=prezzo+totale;
+        }
+        scontoEffettivo=sconto*totale;
+        totaleScontato=totale-scontoEffettivo;
+        rimanenza=portafoglio-totaleScontato;
+
+
+
+
+        if((portafoglio+scontoEffettivo)>totale){
+            //stampa inizio
+
+            var today  = new Date();
+            var options = { weekday: 'short', year: 'numeric', month: 'long', day: 'numeric' };
+            const data=`${today.toLocaleDateString("en-US", options)}`;
+
+            const dataOra=`${today.toLocaleTimeString("en-US",options)}`;
```

Comment:
quante volte dichiari la  data :sweat_smile:?

Benissimo per `toLocaleTimeString` ma usa `it-IT`  => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation